components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt

```diff
@@ -10,6 +10,14 @@ import mozilla.appservices.logins.rust.RustError
 import mozilla.appservices.sync15.SyncTelemetryPing
 import java.util.concurrent.atomic.AtomicLong
 import org.json.JSONArray
+import org.mozilla.appservices.logins.GleanMetrics.LoginsStore as LoginsStoreMetrics
+
+/* I do not like importing these private classes, but
```

Comment:
At the beginning I was like "NOOOOOOOOOO" :-) Reading through the code again, it feels ok: you're just using these to define the "types", you're not instantiating Glean types manually.

We are really trying hard to prevent people to create instances of Glean SDK types at run-time, as that would circumvent our build time checks. Unfortunately, we can't make the constructor private/internal because the generated APIs need to be able to instantiate them :)

Moreover, you write `If we're doing this, it's on us to fix any bustage on version updates.` which sounds fair to me :)

Reply:
😆 I will update the comment to explain what we're using them for, to avoid scaring future readers!